### PR TITLE
Bugfix/edit date

### PR DESCRIPTION
**Pull request by chrisjs87**
## Description

**Please note that this branch incorporated changes from another pending PR ([Bugfix/Edit-Case](https://github.com/Lambda-School-Labs/human-rights-first-asylum-fe-a/pull/359)), which were required in order for this PR to function correctly. The only relevant commit is the final commit. See comments in code indicating what changes were made in this pull request. The only files that were changed were EditCaseDetails and format_date_utils. The three others can be ignored.**

Fixed a bug where the edit case date was not initially populating when the edit case form mounted. We were able to fix this by changing the format of the date data being used. We also needed to update the form's date input to be named correctly and use the updated decision_date variable. This has restored functionality to the decision date detail for cases to be edited and viewed dynamically.

Co-authored by:
@schandrupatla
@joshuascan
@chrisjs87 

[Trello](https://trello.com/c/ShSBhUVr)

https://user-images.githubusercontent.com/75971516/133849089-7d511c58-39a5-4a02-bf06-67597833343f.mp4

Fixes # (issue)

-Fixes issue with being able to edit a case's decision date

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/pages/CaseOverview/EditCaseDetails.js [hunk 1]
```diff
@@ -109,10 +103,10 @@ const EditCaseDetails = props => {
           <Input
             id="date"
             type="date"
-            name="date"
+            name="decision_date"
```
Comment (chrisjs87) [inline src/components/pages/CaseOverview/EditCaseDetails.js:106]: We made sure to line the name up with the relevant schema name/variable

File: src/components/pages/CaseOverview/EditCaseDetails.js [hunk 2]
```diff
@@ -109,10 +103,10 @@ const EditCaseDetails = props => {
...
             onChange={onChange}
             className="text-field"
-            value={formValues.date}
+            value={formValues.decision_date}
```
Comment (chrisjs87) [inline src/components/pages/CaseOverview/EditCaseDetails.js:109]: We made sure to line the name up with the relevant schema name/variable

File: src/components/pages/CaseOverview/EditCaseDetails.js [hunk 3]
```diff
@@ -1,56 +1,45 @@
...
+    caseId,
+    setIsEditModalVisible,
+    isEditModalVisible,
+    setCaseData,
+    caseData,
+    setHasUpdated,
+  } = props;
+
+  const initialFormValues = {
+    decision_date: formatDateYMD(caseData.decision_date),
```
Comment (chrisjs87) [inline src/components/pages/CaseOverview/EditCaseDetails.js:18]: We made sure to line the name up with the relevant schema name/variable

File: src/utils/format_date_util.js [hunk 4]
```diff
@@ -4,3 +4,10 @@ export function formatDate(text) {
   var day = text.slice(8, 10);
   return month + '/' + day + '/' + year;
 }
+
```
Comment (chrisjs87) [inline src/utils/format_date_util.js:7]: Here is the util we added to format the date data in a YYYY-MM-DD format

**Review by zunerabokhari — APPROVED** [review]
This bugfix seems very useful. It completely makes sense to be able to edit the date as an admin. Your video was great. It was so helpful how Sri started the video but showing exactly what was being changed and the before and after. Especially since I'm from the DS side so I might not have fully understood your code but this way I could see the usefulness of this change. The date to decision_date was also nicely explained. And again showing the lines of code you changed and how they're reflected on the front end made everything super clear.

**Review by prodhc — APPROVED** [review]
Great job on this ticket. The description was very accurate and relevant, and the loom video felt professional. Good catch on the unnecessary imports as well. It's nice to see the variable names getting updated throughout the whole application. Going through the code I currently see no apparent errors or mistakes.

**Review by willmoon19 — APPROVED** [review]
This looks like a really good change. I believe updating the names so they match the tables and are more specific is really important. I watched your video and you guys talked about your code very thoroughly and the change works! I see no problems.

**Review by RickyKlusmeier — APPROVED** [review]
This is definitely a good fix and one that was needed so once the data was in there it could be edited if a change happened. I also like how you put the format date in its own utils function so if it were needed elsewhere it could be easily imported.

**Review by jenna-anderson — APPROVED** [review]
The video and description are clear and concise. I love that this problem was solved by the collective knowledge of a backend developer with the alignment of schema, frontend developer with the form functionality, and UX/UI developer with familiarity with how Ant Design wanted the date to be formatted. Go team!

File: src/utils/format_date_util.js (hunk 4 above)
Comment (paulstgermain) [reply src/utils/format_date_util.js:7]: LOVE the addition of a _reusable_ util for formatting dates - something that could occur frequently across the app.

**Pull request merged**